### ISSUE TITLE
Object.basename is deprecated.

### ISSUE THREAD

**Issue by w0k3** [kilian-ito/atom-npm-outdated#5]
Arguments to `path.basename` must be strings
```
Object.basename (/opt/atom/resources/app/src/electron-shims.js:31:10)
NpmWorker.isTargettingFile (/home/dreamy/.atom/packages/atom-npm-outdated/lib/worker.js:270:32)
NpmWorker.link (/home/dreamy/.atom/packages/atom-npm-outdated/lib/worker.js:386:17)
Object.getSuggestionForWord (/home/dreamy/.atom/packages/atom-npm-outdated/lib/index.js:91:33)
<unknown> (/home/dreamy/.atom/packages/atom-ide-ui/modules/atom-ide-ui/pkg/hyperclick/lib/Hyperclick.js:110:35)
Generator.next (null:null:null)
```

**Comment by kilian-ito** [comment]
Hi @w0k3 ,

I close the issue because I cannot reproduce

Regards

**Comment by phyllisstein** [comment]
I just saw the same issue in Atom v1.29.0-beta1. Here's the stacktrace:

```
Object.basename - /Applications/Atom Beta.app/Contents/Resources/app.asar/src/electron-shims.js:31:10
NpmWorker.isTargettingFile - /Users/daniel/.atom/packages/atom-npm-outdated/lib/worker.js:266:32
NpmWorker.link - /Users/daniel/.atom/packages/atom-npm-outdated/lib/worker.js:386:17
Object.getSuggestionForWord - /Users/daniel/.atom/packages/atom-npm-outdated/lib/index.js:97:33
Hyperclick.getSuggestion - /Users/daniel/.atom/packages/atom-ide-ui/modules/atom-ide-ui/pkg/hyperclick/lib/Hyperclick.js:111:33
<unknown> - null:null:null
```

...and I've Gisted my `package.json`, in case there are any clues in there: <https://gist.github.com/phyllisstein/81e65a83957f67a923134df8faf9abca>. Happy to furnish more color if it's useful! Thanks for your work on such an invaluable package.

**Comment by kilian-ito** [comment]
Hello @phyllisstein ,

Thank you for the details.
The issue was in the case a new file (not saved yet) is open in Atom.

It has been fixed in 0.15.1.

Regards

**Comment by phyllisstein** [comment]
Gotcha! Thanks so much for the quick turnaround.
